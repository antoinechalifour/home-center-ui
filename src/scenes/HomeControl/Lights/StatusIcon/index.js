import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Icon from 'react-icons/lib/md/lightbulb-outline'

export default function StatusIcon ({ isOn, onClick }) {
  return (
    <IconWrapper ison={isOn}>
      <Icon onClick={onClick} />
    </IconWrapper>
  )
}

StatusIcon.propTypes = {
  isOn: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
}

const IconWrapper = styled.span`
  svg {
    cursor: pointer;
    font-size: 50px;
    margin-right: 12px;

    opacity: ${({ theme, ison }) => (ison ? 1 : 0.24)};
    color: ${({ theme, ison }) => (ison ? theme.colors.primary : theme.colors.textInverse)};
  }
`
