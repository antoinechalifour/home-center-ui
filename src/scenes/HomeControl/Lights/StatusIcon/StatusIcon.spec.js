jest.mock('react-icons/lib/md/lightbulb-outline', () => 'Icon')

import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import StatusIcon from './'

const theme = {
  colors: { accent: 'red', text: 'black' }
}

describe('ui/HomeControl/Lights/StatusIcon', () => {
  it('Should render correctly (off)', () => {
    const props = {
      isOn: false,
      onClick: jest.fn()
    }
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <StatusIcon {...props} />
      </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly (on)', () => {
    const props = {
      isOn: true,
      onClick: jest.fn()
    }
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <StatusIcon {...props} />
      </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
  })

  it('Should handle click events', () => {})
})
