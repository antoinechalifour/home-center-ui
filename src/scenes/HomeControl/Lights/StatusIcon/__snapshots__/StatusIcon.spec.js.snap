// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/HomeControl/Lights/StatusIcon Should render correctly (off) 1`] = `
.c0 {
  cursor: pointer;
  font-size: 54px;
  margin-right: 12px;
  opacity: 0.65;
  color: black;
}

<Icon
  className="c0"
  onClick={[Function]}
/>
`;

exports[`ui/HomeControl/Lights/StatusIcon Should render correctly (on) 1`] = `
.c0 {
  cursor: pointer;
  font-size: 54px;
  margin-right: 12px;
  opacity: 1;
  color: red;
}

<Icon
  className="c0"
  onClick={[Function]}
/>
`;
